mod-source-record-storage-server/src/main/java/org/folio/consumers/ParsedRecordChunksKafkaHandler.java

```diff
@@ -114,31 +112,36 @@ private Future<String> sendBackRecordsBatchResponse(RecordsBatchResponse records
     String topicName = KafkaTopicNameHelper.formatTopicName(kafkaConfig.getEnvId(), KafkaTopicNameHelper.getDefaultNameSpace(),
       tenantId, DI_PARSED_RECORDS_CHUNK_SAVED.value());
 
-    KafkaProducerRecord<String, String> targetRecord =
-      KafkaProducerRecord.create(topicName, key, Json.encode(event));
+    var targetRecord =
+      new KafkaProducerRecordBuilder<String, Object>(tenantId)
+        .key(key)
+        .value(event)
+        .topic(topicName)
+        .build();
 
     targetRecord.addHeaders(kafkaHeaders);
 
     Promise<String> writePromise = Promise.promise();
 
     String producerName = DI_PARSED_RECORDS_CHUNK_SAVED + "_Producer";
-    KafkaProducer<String, String> producer =
-      KafkaProducer.createShared(Vertx.currentContext().owner(), producerName, kafkaConfig.getProducerProps());
+    KafkaProducer<String, String> producer = new SimpleKafkaProducerManager(
```

Comment:
Now that i think about it, the SimpleKafkaProducerManager should probably be a property of this class that is created once. Then use `createShared` method to create producers.